src/superqt/cmap/_cmap_utils.py

```diff
@@ -120,6 +120,7 @@ def paintEvent(self, event) -> None:
             gradient.setColorAt(stop.position, QColor(stop.color.hex).lighter(lighter))
         painter.setBrush(gradient)
         painter.drawRect(rect)
+    painter.end()
```

Comment:
I expect that `painter.end()` should be called only if `painter_or_device` is `QPaintDevice` instance. Otherwise, the caller is responsible for close painter. 

It may happen that caller of `draw_colormap` may want to draw something after coloromap, so need active painter instance.

Reply:
Great catch @Czaki, thank you! I've updated the changeset in b53b5ce